fixups/FileRedirectionFixup/PathRedirection.cpp

```diff
@@ -349,6 +339,62 @@ std::wstring RedirectedPath(const normalized_path& deVirtualizedPath, bool ensur
     return result;
 }
 
+/// <summary>
+/// Figures out the absolute path to redirect to.
+/// </summary>
+/// <param name="deVirtualizedPath">The original path from the app</param>
+/// <param name="ensureDirectoryStructure">If true, the deVirtualizedPath will be appended to the allowed write location</param>
+/// <returns>The new absolute path.</returns>
+std::wstring RedirectedPath(const normalized_path& deVirtualizedPath, bool ensureDirectoryStructure)
+{
+    /*
+        To prevent apps breaking on an upgrade we redirect writes to the package root path to
+        a path that contains the package family name and not the package full name.
+    */
+    std::wstring result;
+    bool shouldredirectToPackageRoot = false;
+    if (deVirtualizedPath.full_path.find(g_packageRootPath) != std::wstring::npos)
+    {
+        result = LR"(\\?\)" + g_writablePackageRootPath.native();
+        shouldredirectToPackageRoot = true;
+    }
+    else
+    {
+        //TEST
+        //MessageBoxEx(NULL, L"Redirecting to VFS", L"Redirecting to VFS", 0, 0);
+        result = LR"(\\?\)" + g_redirectRootPath.native();
+    }
+
+    std::wstring_view relativePath;
+    if (shouldredirectToPackageRoot)
+    {
+        auto lengthPackageRootPath = g_packageRootPath.native().length();
+        auto stringToTurnIntoAStringView = deVirtualizedPath.full_path.substr(lengthPackageRootPath);

```

Comment:
No need to swap from a string to a string_view. Change GenerateRedirectedPath to take in a wstring ref instead of a string_view OR inline the string_view transformation in the statement here and remove the declaration above (std::wstring_view relativePath)